menhir-recover/main.ml

```diff
@@ -67,6 +75,62 @@ let () =
       );
   end
 
+let () =
```

Comment:
I would extract all of that into a separate file and split it into functions like "print reduction", "print transaction", "print incoming" and so on.

Reply:
done